framework/auth/views.py

```diff
@@ -207,6 +230,69 @@ def forgot_password_post():
     return {}
 
 
+def forgot_password_institution_post(mail_template=mails.FORGOT_PASSWORD_INSTITUTION,
```

Comment:
As mentioned before, merging them into one template and letting arguments control the behavior is a probably the better way to go.